project_1/include/simulator.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include "utils.hpp"
+
+#include <vector>
+#include <cstdint>
+#include <queue>
+
+class NetworkSimulator {
+private:
+    std::uint32_t bandwidth = 52428800;
+    std::uint32_t data_transmission_frequency_bytes = 52428800;
+    std::uint32_t data_transmission_frequency_packets = 51200;
+    std::uint32_t packet_size = 1024;
+
+    // Contains all sending servers 
+    std::vector<ServerSender> servers{};
+    ServerBase server_receiver{};
+
+    // Sorts packets by travel time
+    std::priority_queue<PacketHeader> packets{};
+    
+    // Contains all events 
+    std::vector<Event> events{};
+
+public:
+
+    explicit NetworkSimulator(const std::vector<std::uint32_t>&);
```

Comment:
In my opinion it will be better to add parameter names in constructors, without them it is hard to deduce what is expected as constructor parameters or in which order without looking in source .cpp file
```suggestion
    explicit NetworkSimulator(const std::vector<std::uint32_t>& /* distances_to_receiver */);
```

Reply:
Point holds for all constructors